### ISSUE TITLE
Reduce bus factor for service integrations.

### ISSUE THREAD

**Issue by acrymble** [programminghistorian/jekyll#849]
At the moment a number of our services are tied to people's personal email accounts.

https://github.com/programminghistorian/jekyll/wiki/Service-Integrations

These need to be centralized into "Programming Historian" controlled accounts in case anyone leaves or otherwise gets hit by a bus.

**Comment by drjwbaker** [comment]
Is there a downside of having an proghist email address (eg gmail) that we use for these things? (especially, if we don't expose that email anywhere online as a point of contact, therefore don't need to check, and setup forwarding redirects to our group chat)

**Comment by acrymble** [comment]
is it possible for it to be a @programminghistorian.org email instead? Or do we lack that ability?

**Comment by drjwbaker** [comment]
Custom email domains are usually linked to custom web domains, so CHNM might be able to help here.

**Comment by drjwbaker** [comment]
Or we could just pay for it.

**Comment by acrymble** [comment]
List of services linked to personal accounts:

- Directory of Open Access (DOAJ) @drjwbaker 
- Twitter @jerielizabeth 
- Heroku @walshbr 
- Google sheets keychain @walshbr 
- Google Analytics @acrymble 
- Cloudflare @mdlincoln
- Netlify @mdlincoln
- Zenodo @drjwbaker

**Comment by mdlincoln** [comment]
It is easy to add more administrative accounts to Netlify.

Transferring a cloudflare site is a bit more involved as we'd need to have CHNM update our nameservers again.

**Comment by drjwbaker** [comment]
Is the action here to take a look at how easily we can reduce the bus factor for the logins we 'own', or to investigate moving to a generic email address?

**Comment by acrymble** [comment]
second option solves first one too, no?

How about we adopt @drjwbaker gmail suggestion and then physically mail out the password details to the editorial board so they can keep them safe on a bulletin board or something?

**Comment by drjwbaker** [comment]
It does. Just trying to avoid duplicating effort.

Send round a wetransfer file with a short expiry date?

**Comment by acrymble** [comment]
~~I've asked Jeri for access to the gmail account.~~

**Comment by acrymble** [comment]
I've set up a new gmail account:

programminghistorian@gmail.com

The following services have been shifted to this account, and the details updated on the Wiki:

- [x] Directory of Open Access (DOAJ) @drjwbaker 
- [x] Twitter @jerielizabeth 
- [x] Google Group email @acrymble
- [x] Heroku @walshbr 
- [x] Google sheets keychain @walshbr 
- [x] Google Analytics @acrymble 
- [ ] Cloudflare @mdlincoln
- [ ] Netlify @mdlincoln
- ~~Zenodo @drjwbaker~~

**Comment by acrymble** [comment]
Can I please ask @drjwbaker @walshbr and @mdlincoln to shift the services they control to the above address and tick off the list above?

**Comment by drjwbaker** [comment]
Done for DOAJ. Question is now, where to store the passwords?

On Zenodo, the process doesn't connect it to any specific account. Anyone of us can follow the process at https://github.com/programminghistorian/jekyll/wiki/Service-Integrations to make the annual deposit (the only downside of which is that only the person who made the deposit can edit the deposit, but I can't see that being necessary years after the fact). To do a Zenodo deposit through a Zenodo account with programminghistorian@gmail.com as login would also require us having a GitHub account with programminghistorian@gmail.com as associated email to the login. Which I presume we don't want to do.

**Comment by walshbr** [comment]
Could you use the Programming Historian's GitHub organization for Zenodo (not sure what email is associated with that)? Or does it have to be a user account?

I'll take a look at heroku and google sheets - it might take some tweaking to the bot script to update the credentials, and I won't have time to take a look at this for a week or so.

**Comment by acrymble** [comment]
Is Github one of our services we need to list too? @walshbr. I didn't think about that one.

**Comment by acrymble** [comment]
I realize the bus factor is getting worse until I communicate the password to everyone.

@drjwbaker are you in your office today? I can call you and then let you know verbally so you can share with Anna Maria, and we can disseminate further at the Skype call.

**Comment by acrymble** [comment]
Our twitter account has the following approved apps. Are any of these vital before I remove them?
<img width="599" alt="screen shot 2018-05-22 at 21 38 09" src="https://user-images.githubusercontent.com/646647/40471693-57ea9d5c-5f2f-11e8-881e-21df030a732c.png">

**Comment by acrymble** [comment]
@drjwbaker now knows the password and will tell @amsichani .

**Comment by amsichani** [comment]
@acrymble @drjwbaker looking forward to hear the top secret password. Privacy and security protocols can't really beat the good old-school way ;-)

**Comment by walshbr** [comment]
@acrymble - ProgHistBot is the twitter bot, so don't revoke that. I'm not sure about the other ones. It's been a while since I've made a GitHub organization, but I believe you have to give a billing email address (even if you're a free account). Presumably we have someone's associated with the organization, but I can't see the settings to be able to tell.

**Comment by acrymble** [comment]
@wcaleb originally set up our Github. It must be connected to his account.

**Comment by drjwbaker** [comment]
> Could you use the Programming Historian's GitHub organization for Zenodo (not sure what email is associated with that)? Or does it have to be a user account?

No. The way it works is that in Zenodo you link your Zenodo account to a GitHub account so that Zenodo can 'see' all the GitHub repos that the linked GitHub can make releases of (that is, that account has the GitHub privileges to do so).

**Comment by wcaleb** [comment]
@acrymble As far as I can tell, I've been removed from the PH organization, so I don't have access to its settings anymore and it's no longer connected to my account.

**Comment by acrymble** [comment]
Thanks for clarifying @wcaleb.

**Comment by drjwbaker** [comment]
> @drjwbaker now knows the password and will tell @amsichani 

@amsichani has the password.

**Comment by amsichani** [comment]
we still don’t know the passwords for a couple of services, such as Github, mailing list.

**Comment by walshbr** [comment]
I just signed programminghistorian@gmail.com up for a Heroku account. I don't have the password yet, though, and I'll need to get it before I can continue with the sign up and transfer of the Twitter bot. Anyone up for a thirty-second Skype call?

**Comment by acrymble** [comment]
sure. one sec.

**Comment by walshbr** [comment]
Got the password - I'll try to take a look at Heroku and the Google Sheets setups this afternoon and switch everything over. Should hopefully be a quick fix, but wanted to make sure I had time in case something got weird with deploying.

**Comment by walshbr** [comment]
Changed over the Heroku account and added the gmail account as a co-owner on the google drive app that is running the bot. I think that is what you meant by google sheets keychain @acrymble, but let me know if I misinterpreted. I also transferred ownership of the Twitter spreadsheet itself to the gmail. 

Our joint gmail account should be capable of generating another keychain in the future, but we need to document how to do so on the proghistbot repo documentation. I'll open a separate issue for that. [update: opened and closed!] Another option would be to put the keychain up on our google drive for storage, but not sure if there are security issues with that…

Also would second the suggestion that we make sure this email address is forwarding to the group email. We'd want to know if any of the service change their terms to the extent that we won't be able to use them with enough time to make sure we find alternatives.

**Comment by acrymble** [comment]
@walshbr Good point about monitoring emails. I wonder, do we want to make this the managing editor email, so that person could forward relevant warnings to the appropriate person, rather than add to the email inboxes of the entire group? @JMParr would you be happy to have this box forwarded to you while you hold that role? Or do you prefer not?

**Comment by JMParr** [comment]
Sure.

**Comment by drjwbaker** [comment]
Thanks @JMParr. If you decide it is appropriate to reroute emails to the shared gmail to our googlegroup list, let us know.

**Comment by acrymble** [comment]
@JMParr I have enabled forwarding. I had a look at the account and it's receiving a lot of promotional spam from Heroku. @walshbr could you make sure that isn't clogging up Jess's email? On a related note, @JMParr please do let us know if any of our services are spamming you and the maintainer will make sure it stops.

**Comment by acrymble** [comment]
Ok, I'd like to close this soon. A few outstanding items:

1) @walshbr, @drjwbaker and @mdlincoln can you please make sure the wiki is up to date and all accounts linked to the new google account now explicitly say so.
2) Will we be able to recover all passwords with the information we have, or do we need to share those/standardize those too?

**Comment by walshbr** [comment]
The Heroku email settings should be updated now. Updated the wiki as well.

The Heroku password at least has different character requirements than gmail, so we'd have to modify the original to standardize. I checked, and we should just be able to recover the password via a link emailed to the programminghistorian gmail account.

**Comment by drjwbaker** [comment]
Done

**Comment by acrymble** [comment]
Just a reminder to @mdlincoln this is just waiting for your input.

**Comment by acrymble** [comment]
@mdlincoln can you add the Google account to the Netlifly account as an administrator programminghistorian@gmail.com 

I think we can then close this issue.

**Comment by acrymble** [comment]
@mdlincoln this is waiting on you. Do you need help?

**Comment by mdlincoln** [comment]
Added.
